### ISSUE TITLE
[Bug] UWP Immediate crash on startup in release mode from 4.3 all the way up to 4.5-pre1

### ISSUE THREAD

**Issue by MitchBomcanhao** [xamarin/Xamarin.Forms#9202]
### Description
A blank UWP project with a single added package reference to SkiaSharp.
When built in **Release** mode, it will crash on launch.
It will run if built in **Debug** configuration.

### Steps to Reproduce

1. Open the attached project
2. Build and run the UWP app in Debug mode (it'll work)
3. Now switch to Release mode and run.

### Expected Behavior
The app will launch.
### Actual Behavior
The app will crash on launch/activation
### Basic Information

- Version with issue: all from 4.3 up to latest 4.5-pre1
- Last known good version: 4.2.0.910310
- IDE: VS 2019 16.4.2
- Platform Target Frameworks: 
  - UWP:  17763
- Nuget Packages:  
  - SkiaSharp 1.68.1.1
- Running on Windows 10 build 18363.535

### Reproduction Link
[ReproAppStartCrash.zip](https://github.com/xamarin/Xamarin.Forms/files/4059537/ReproAppStartCrash.zip)

### Workaround
Installing Microsoft.UI.Xaml to **2.3.191211002** (which overrides the implicit **2.1.190606001** version) appears to make it work.

**Comment by MitchBomcanhao** [comment]
I'm not entirely sure what part the SkiaSharp package is playing here.
I made the basic project and it did appear to run. When I added SkiaSharp, it stopped running as expected and I thought I had found the issue, but removing SkiaSharp from the project would still fail to run afterwards - perhaps some sort of reference being left over?

Also not sure of the impact of upgrading Microsoft.UI.Xaml, so I'm only mentioning it as it is relevant to the issue at hand, not promising it as an actual fix that works for everyone.

**Comment by joshminor** [comment]
I am experiencing the same issue, updating the Microsoft.UI.Xaml does seem to work.

**Comment by MitchBomcanhao** [comment]
Updating Microsoft.UI.Xaml also has the negative side effect of changing how some controls are rendered. I had changes in app.xaml to override some default styles, but now those overrides don't work. "I'll fire up the app and use the Live Visual Tree feature in visual studio to check the UI elements", or so I thought - this doesn't appear to work anymore, and inspecting the apps UI only returns the main page object. useless. not sure when this became broken as I've just recently updated to 16.4.2.

**Comment by PureWeen** [comment]
@MitchBomcanhao can you expand on this

> Updating Microsoft.UI.Xaml also has the negative side effect of changing how some controls are rendered.

Do you have a repro we can look at to see what's different?

**Comment by MitchBomcanhao** [comment]
Things I've found:

#############
**Tabbed page**

The standard tabbed page has a margin around its pivot items.
with the default microsoft.ui.xaml (2.1), I can override this margin by adding a thickness with the correct name to the UWP project app.xaml file
`<Thickness x:Key="PivotItemMargin">0,0,0,0</Thickness>`

Once updated to Microsoft.UI.Xaml 2.3, that override is no longer effective, even though the 2.3 package seems to use the exact same name for the pivot item margin - haven't quite figured out what is going wrong.
What I did find is that I can get my margin back to zero by using this style
`<Style TargetType="PivotItem"> <Setter Property="Margin" Value="0" /> </Style>`

here are a couple of screenshots of the app with 2.1 and 2.3
![image](https://user-images.githubusercontent.com/31066938/72602041-8318fe00-390e-11ea-91ca-79409e9752d8.png)
![image](https://user-images.githubusercontent.com/31066938/72602058-87ddb200-390e-11ea-9dde-c3b618fbaf92.png)


#############
**Switch control**

With ui xaml 2.3, the switch control renders incorrectly when toggled = true and hovered with the mouse. Its main color becomes transparent. 
This is not an issue when running a native UWP app with ui.xaml 2.3, only on Xamarin forms apps.
Downgrade the package to 2.1 and the switch works as expected.

![ezgif-3-150a820fd913](https://user-images.githubusercontent.com/31066938/72602538-57e2de80-390f-11ea-8ac7-972277b99beb.gif)


#############
Sample project
[UiXamlBugs.zip](https://github.com/xamarin/Xamarin.Forms/files/4076229/UiXamlBugs.zip)

**Comment by charlesroddie** [comment]
Cc @mattleibow as this concerns skiasharp.

**Comment by charlesroddie** [comment]
@foggyfinder could you check if this causes a problem in our app?

**Comment by FoggyFinder** [comment]
> A blank UWP project with a single added package reference to SkiaSharp.

@MitchBomcanhao In your MCVE there is no reference to XF inside of UWP project. Add one and, I suppose, the issue will be resolved.

**Comment by MitchBomcanhao** [comment]
Following advice from @FoggyFinder, I've tried a couple of things...

- First, went back to the broken state without the Microsoft.UI.Xaml change (deleting the solution folder and reopening the zipped repro project).
- then added the **Xamarin.Forms** reference to the UWP project - this apparently made it work.
- went back to the broken state (deleting and redeploying repro project)
- Added a **SkiaSharp.Views.Forms** reference to the UWP project. this also made it work - maybe because it has an explicit dependency on Xamarin.forms.

ok, so it seems that when native compilation is used (release mode), there's some issue with references and it struggles to compile correctly, leading to it not running correctly. This was not an issue with previous versions of Xamarin.Forms and UWP, so something must have changed.

I've attempted similar package reference changes on our production solution, so far without success.

**Comment by FoggyFinder** [comment]
> This was not an issue with previous versions of Xamarin.Forms and UWP, so something must have changed.

This was the same in previous versions of XF as well. You can try downgrading version of package in your test project and you see no difference.

> I've attempted similar package reference changes on our production solution, so far without success.

Well, I suppose because the root of the matter is somewhere else. Good luck.

**Comment by MitchBomcanhao** [comment]
I've found a commit that references a few of the things that are surfacing here:
[https://github.com/xamarin/Xamarin.Forms/commit/f51099c3823cd7433d1fa95c4cff9227b2a59629](https://github.com/xamarin/Xamarin.Forms/commit/f51099c3823cd7433d1fa95c4cff9227b2a59629)

This was done only a few months ago, and must have been introduced to Xamarin forms 4.3 (where issues started for me).

It adds a direct dependency to 
`<dependency id="Microsoft.UI.Xaml" version="2.1.190606001" />`

in Xamarin.Forms.Platform.UAP/Forms.cs, there's a new bit of code that attempts to add XamlControlsResources to the current app resources.
```
Windows.UI.Xaml.Application.Current.Resources.MergedDictionaries.Add(GetTabletResources());

try
{
	Windows.UI.Xaml.Application.Current.Resources.MergedDictionaries.Add(new Microsoft.UI.Xaml.Controls.XamlControlsResources());
}
catch
{
	Log.Warning("Resources", "Unable to load WinUI resources. Try adding Xamarin.Forms nuget to UWP project");
}
```
Now, when we manually add the Microsoft.UI.Xaml package to the UWP project, it mentions this in the readme: 
```
Thanks for installing the WinUI nuget package! Don't forget to add this to your app.xaml:

    <Application.Resources>
        <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls"/>
    </Application.Resources>
```
so, should we be merging that dictionary in the app.xaml file, or should Xamarin forms be doing it for us? 

there's clearly something new here, and I'd say that's the bit that is failing, because fiddling with the microsoft.ui.xaml references has a direct effect on this. I've seen references to updating microsoft.ui.xaml in other recent Xamarin forms GitHub issues about UWP apps crashing on launch.

**Comment by PureWeen** [comment]
I'm going to close this issue for now

I moved the layout related issues over here https://github.com/xamarin/Xamarin.Forms/issues/9202

As far as this issue what I can tell is this
- If you add SkiaSharp.Views.Forms to the UWP platform project it runs fine maybe there's something here that @mattleibow  needs to fix?
- If you bump WinUI to 2.3 it works. There was a fix that went into WinUI 2.2 that fixed some transitive behaviors of WinUI https://github.com/microsoft/microsoft-ui-xaml/pull/1203 and we are going to be bumping up to the latest WinUI version with XF 4.5 https://github.com/xamarin/Xamarin.Forms/pull/9132

**Comment by KarolStosikPowel** [comment]
Adding Microsoft.UI.Xaml nuget breaks showing popups on UWP. Adding SkiaSharp.Views.Forms doesn't fix anything in my case. I guess Microsoft.UI.Xaml is included in  Xamarin.Forms 4.5.0.282-pre4 because popups are also broken there.
